kimchi/src/circuits/constraints.rs

```diff
@@ -146,6 +148,10 @@ pub struct ConstraintSystem<F: FftField> {
     #[serde_as(as = "o1_utils::serialization::SerdeAs")]
     pub endomul_scalar8: E<F, D<F>>,
 
+    /// Foreign field multiplication gate selector polynomials
+    #[serde(bound = "Vec<foreign_mul::SelectorPolynomial<F>>: Serialize + DeserializeOwned")]
+    pub foreign_mul_selector_polys: Vec<foreign_mul::SelectorPolynomial<F>>,
```

Comment:
Yes, exactly.   I decided to make it a vector of selector polynomials rather than an optional object containing arrays of polynomials.  This field is set by calling `foreign_mul::selector_polynomials(...)`.  When the vector is empty is means there are no foreign multiplication gates.